transaction/core/src/onetime_keys.rs

```diff
@@ -178,126 +213,243 @@ mod tests {
     use mc_crypto_rand::McRng;
     use mc_util_from_random::FromRandom;
 
+    // Returns (onetime_public_key, tx_public_key)
+    fn get_output_public_keys(
+        tx_private_key: &RistrettoPrivate,
+        recipient: &PublicAddress,
+    ) -> (RistrettoPublic, RistrettoPublic) {
+        let onetime_public_key = create_onetime_public_key(&tx_private_key, recipient);
+        let tx_public_key = create_tx_public_key(&tx_private_key, recipient.spend_public_key());
+        (onetime_public_key, tx_public_key)
+    }
+
+    // Get the account's i^th subaddress.
+    fn get_subaddress(
+        account: &AccountKey,
+        index: u64,
+    ) -> (RistrettoPrivate, RistrettoPrivate, PublicAddress) {
+        // (view, spend)
+        let (c, d) = (
+            account.subaddress_view_private(index),
+            account.subaddress_spend_private(index),
+        );
+
+        // (View, Spend)
+        let (C, D) = (RistrettoPublic::from(&c), RistrettoPublic::from(&d));
+        // Look out! The argument ordering here ie weird.
+        let subaddress = PublicAddress::new(&D, &C);
```

Comment:
It's a bummer that PublicAddress::new takes the spend key before the view key, which leads to this counter-intuitive order. C and D have the same type, so I'm afraid to just swap the arguments.

Reply:
There are only a few (3?) places in the code where we explicitly use `::new()` but swapping the order would render all the testnet utxos unspendable... maybe we should fix this for main net as a break change and add a stronger type wrapper on `spend` and `new` at the same time...